llsd/serde_binary.py

```diff
@@ -7,6 +7,14 @@
                        _str_to_bytes, binary, is_integer, is_string, uri)
 
 
+try:
+    # Python 2
+    xrange
+except NameError:
+    # Python 3
+    xrange = range
+
+
```

Comment:
/me shakes fist at python 2

Reply:
Agreed ... but `llsd` is perhaps _the_ most ubiquitous of Linden's Python packages, and it's certain that there are still Python 2 consumers. I don't want any obstacles of the form: "We'd have to update to Python 3 before we could switch to the newer python-llsd implementation."

Reply:
yeah not trying to derail things too much, I just wish we could pretend to be more like python3, something like

```
try:
    range = xrange  # pretend to be python 3
except NameError:
    # hooray we are!
   pass
```
and then we could use proper `range()` everywhere below

Reply:
Okay: https://github.com/secondlife/python-llsd/pull/6/commits/9249fcee5c66b4f44d602eafd09569806d78195e